pineapple-bundle/pineapple-core/src/main/java/sh/miles/pineapple/config/type/Configuration.java

```diff
@@ -0,0 +1,234 @@
+            field.setVisible();
+            String path = field.getPath();
+
+            boolean mapOnly = field.getRuntimeClass() == Map.class && path.isEmpty() && this.fields.size() == 1;
+
+            if (!path.isEmpty()) {
+                config.setComments(path, field.getComments());
+            }
+
+            TypeAdapter<Object, Object> adapter = (TypeAdapter<Object, Object>) configManager.getTypeAdapter(ConfigType.get(field.getField()));
+            if (adapter == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Unable to find type adapter for field: {0} with path: {1} in {2}", objArr(field.getField().getName(), field.getPath(), this.file.getName()));
+                field.setHidden();
+                continue;
+            }
+
+
+            Object toSave = this.getValue(field.getField());
+            if (toSave == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Found null value while trying to save for field name: {0}, path: {1} in config {2}", objArr(field.getField().getName(), path, this.file.getName()));
+            }
+
+            Object existing = config.get(path);
+            if (existing != null) {
+                if (!replace) {
+                    continue;
+                }
+            }
+
+            if (existing instanceof ConfigurationSection section) {
+                existing = section.getValues(false);
+            }
+
+            Object value = adapter.write(toSave, existing, replace);
+            if (value == null) {
+                PineappleLib.getLogger().log(Level.WARNING, "Found null value from TypeAdapter: {0} while trying to save config {1}", objArr(adapter.getClass().getTypeName(), this.file.getName()));
+                field.setHidden();
+                continue;
+            }
+
+            if (mapOnly) {
+                Map<String, Object> saving = (Map<String, Object>) value;
+                for (Map.Entry<String, Object> entry : saving.entrySet()) {
+                    config.set(entry.getKey(), entry.getValue());
+                }
+                field.setHidden();
+                break;
+            }
+
+
+            config.set(path, value);
+            field.setHidden();
+        }
+
+        try {
+            config.save(this.file);
+        } catch (IOException ex) {
+            PineappleLib.getLogger().log(Level.WARNING, ex, () -> "Unable to save config %s".formatted(this.file.getName()));
+        }
+        return this;
+    }
+
+    private void setField(Field field, Object value) {
+        this.fieldModifier.setField(field, value, instance);
+    }
+
+    private Object getValue(Field field) {
+        return this.fieldModifier.getValue(field, instance);
+    }
+
+
+    private boolean setupFile() {
+        try {
+            if (this.file.isDirectory()) {
+                return false;
+            }
+
+            if (!this.file.getParentFile().exists() && !this.file.getParentFile().mkdirs()) {
```

Comment:
Doesn't necessarily look as beautiful but we can afford an or merge on the last 2 if statements